File: src/be/src/WHO.BioHub.Models/Models/Resource.cs
using WHO.BioHub.Models.Models.Abstractions;
using WHO.BioHub.Shared.Enums;

namespace WHO.BioHub.Models.Models;

public class Resource : EntityBase
{
    public string Name { get; set; }
    public string Extension { get; set; }    
    public DateTime? UploadTime { get; set; }
    public Guid? UploadedById { get; set; }
    public virtual User UploadedBy { get; set; }    
    public ResourceType Type { get; set; }
    public ResourceFileType? FileType { get; set; }
    public bool? Current { get; set; }    
    public Guid? ParentId { get; set; }   
}

File: src/fe/src/models/Resource.ts
import { ResourceFileType } from "./enums/ResourceFileType";
import { ResourceType } from "./enums/ResourceType";

export interface Resource {
  Id: string;
  Name: string;
  Extension: string | undefined;
  Type: ResourceType;
  FileType: ResourceFileType | undefined;
  Current: boolean | undefined;
  UploadTime: Date;
  UploadedBy: string;
  ParentId: string;
}
